catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java

```diff
@@ -1644,6 +1685,232 @@ public void setMetacardTypes(List<MetacardType> types) {
+            throw new IngestException("Unsupported Query when querying for transaction results. ",
+                    e);
+        }
+
+        return new CreateResponseImpl(createRequest, properties, createdMetacards);
+    }
+
+    @Override
+    public UpdateResponse update(UpdateRequest updateRequest) throws IngestException {
+        Map<String, Serializable> properties = new HashMap<>();
+
+        Subject subject =
+                (Subject) updateRequest.getPropertyValue(SecurityConstants.SECURITY_SUBJECT);
+        Csw csw = factory.getClientForSubject(subject);
+        CswTransactionRequest transactionRequest = new CswTransactionRequest();
+        transactionRequest.setVersion(CswConstants.VERSION_2_0_2);
+        transactionRequest.setService(CswConstants.CSW);
+        transactionRequest.setVerbose(true);
+
+        OperationTransaction opTrans = (OperationTransaction) updateRequest.getPropertyValue(
+                Constants.OPERATION_TRANSACTION_KEY);
+
+        String insertTypeName = schemaTransformerManager.getTransformerIdForSchema(
+                cswSourceConfiguration.getOutputSchema());
+
+        if (insertTypeName == null) {
+            insertTypeName = CswConstants.CSW_RECORD;
+        }
+
+        ArrayList<Metacard> updatedMetacards = new ArrayList<>();
+
+        ArrayList<Filter> updatedMetacardFilters = new ArrayList<>();
+
+        for (Map.Entry<Serializable, Metacard> update : updateRequest.getUpdates()) {
+            Metacard metacard = update.getValue();
+            properties.put(metacard.getId(), metacard);
+            updatedMetacardFilters.add(filterBuilder.attribute(updateRequest.getAttributeName())
+                    .is()
+                    .equalTo()
+                    .text(metacard.getAttribute(updateRequest.getAttributeName())
+                            .getValue()
+                            .toString()));
+            transactionRequest.getUpdateActions()
+                    .add(new UpdateAction(metacard, insertTypeName, null));
+        }
+        try {
+            TransactionResponseType response = csw.transaction(transactionRequest);
+            if (response.getTransactionSummary()
+                    .getTotalUpdated()
+                    .longValue() != updateRequest.getUpdates()
+                    .size()) {
+                throw new CswException("One or more updates failed");
+            }
+
+        } catch (CswException e) {
+            LOGGER.debug("Csw Transaction Failed : " + e.getMessage());
+            throw new IngestException("Csw Transaction Failed : " + e.getMessage());
+        }
+
+        try {
+            updatedMetacards.addAll(transactionQuery(updatedMetacardFilters));
+        } catch (UnsupportedQueryException e) {
+            LOGGER.debug("Unsupported Query when querying for transaction results. ", e);
+            throw new IngestException("Unsupported Query when querying for transaction results. ",
```

Comment:
Same as above. At this point, some/all of the metacards have been updated and instead of an exception, an error report should be added to the response.